File: pandascript-vs-cdp/scripts/pandascript/news.js
const page = new Page();
await page.goto("https://apnews.com/hub/world-news");

const { links } = page.extract({
  links: [{ selector: "a[href*='/article/']", attr: "href", limit: 10 }]
});
const urls = [...new Set(links)].slice(0, 3);

const articles = [];
for (const url of urls) {
  await page.goto(url);
  page.waitForSelector(".RichTextStoryBody p");
  const article = page.extract({
    headline: "h1",
    paragraphs: [{ selector: ".RichTextStoryBody p", limit: 3 }]
  });
  articles.push({ url, headline: article.headline, paragraphs: article.paragraphs });
}

return articles;

File: pandascript-vs-cdp/scripts/lightpanda-py/news.py
import json
import os

from lightpanda import Browser

args = os.environ.get("BENCH_LPD_ARGS", "").split()

with Browser(args=args) as b:
    page = b.new_session()
    page.goto(url="https://apnews.com/hub/world-news")

    links = page.extract(schema={
        "links": [{"selector": "a[href*='/article/']", "attr": "href", "limit": 10}],
    })["links"]
    urls = list(dict.fromkeys(links))[:3]

    articles = []
    for url in urls:
        page.goto(url=url)
        page.wait_for_selector(selector=".RichTextStoryBody p")
        article = page.extract(schema={
            "headline": "h1",
            "paragraphs": [{"selector": ".RichTextStoryBody p", "limit": 3}],
        })
        articles.append({"url": url, "headline": article["headline"],
                         "paragraphs": article["paragraphs"]})

print(json.dumps(articles))
